### ISSUE TITLE
Feature request: Warn users when adding multiple values to dimension before flushing

### ISSUE THREAD

**Issue by allenheltondev** [aws-powertools/powertools-lambda-typescript#3350]
### Use case

In reference to [this post](https://bsky.app/profile/did:plc:aqtn3pn3dk4kd76fleosrrxb/post/3lbkgd7kpik2q), when a user adds multiple values to a single dimension before flushing metrics, only the last value set on the dimension wins.

### Solution/User Experience

It would be nice if the metrics object would do a `console.warn` when multiple distinct values are used on a single dimension before flushing. Current behavior uses the last value added to the dimension.

### Alternative solutions

```Markdown

```

### Acknowledgment

- [x] This feature request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

**Comment by dreamorosi** [comment]
Hi @allenheltondev, nice to see you here!

I agree with the discussion you and @leandrodamascena had in the linked thread, we should emit a warning when this happens.

I have opened a PR to introduce the change, once merge it'll be included in the next release.
